pkg/deployments/zeroscaler/zeroscaler.go

```diff
@@ -131,17 +131,35 @@ func (z *zeroscaler) ensureMetricsCollection(deployment *appsv1.Deployment) {
 		if ok {
 			collector.stop()
 		}
+		metricsCheckInterval, err := k8s.GetMetricsCheckInterval(
+			deployment.Annotations,
+		)
+		if err != nil {
+			glog.Warningf(
+				"Invalid custom metrics check interval value in deployment %s,"+
+					" falling back to the default value of %d seconds; error: %s",
+				deployment.Name,
+				z.cfg.MetricsCheckInterval,
+				err,
+			)
```

Comment:
The error you return from `GetMetricsCheckInterval()` already indicates it was an issue with paring the value. Since you're including the text of the error at the end of the warning message, this information basically appears twice.

I would refrain from mentioning the value as "invalid" here and just note "there was an error getting.... falling back to." This way, also, if `GetMetricsCheckInterval()` ever changes and produces _other_ possible errors as well, there won't be text elsewhere in the code (e.g. right here) that needs to be adjusted to account for the different sorts of errors that function can produce.